### ISSUE TITLE
to do: Implement Tests

### ISSUE THREAD

**Issue by dario-piotrowicz** [dario-piotrowicz/ReactJS_e-commerce-miniproject#8]
Implement some tests using [`jest`](https://jestjs.io/) and [`enzyme`](https://enzymejs.github.io/enzyme/) (the former is already included thanks to create-react-app, the latter needs to be installed).

Don't go overboard as those are not even shipped to production nor we actually need to test anything in this miniproject, just add some as a coding exercise.

**Comment by dario-piotrowicz** [comment]
Note: Seems that from Jest v24 snapshot made from Enzyme's shallowWrappers don't work anymore

To make them work a snapshotSerialized has to be provided ( for example I tried with [enzyme-to-json](https://github.com/adriantoine/enzyme-to-json) and it did work as expected, the problem being that did not work when running coverage tests )

**Comment by dario-piotrowicz** [comment]
I managed to make the shallowWrappers work, but they result empty if running the tests with `--coverage`.

I tried all sorts of things (including [React's Shallow Renderer](https://reactjs.org/docs/shallow-renderer.html)) but could not manage to get the shallow rendered component working when checking the tests' coverage.

Since even React's official shallow renderer had the same issue I believe this cannot be helped (is it a bug? something no one cares about?), but I do not want to implement something with such hazy behaviour. 
So eventually I decided no to implement snaphot testing nor using enzyme.

So (unfortunately) in the solution I will adapt only standard jest unit tests will be used.
